<a href="https://colab.research.google.com/github/KhyatiMahendru/HappinessIndexAnalysis/blob/master/DelhiSurveyAnalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [44]:
from google.colab import files
uploaded = files.upload()

Saving Happiness index - Form Responses 1.csv to Happiness index - Form Responses 1 (1).csv


In [0]:
import io
data = pd.read_csv(io.BytesIO(uploaded['Happiness index - Form Responses 1.csv']))

In [46]:
data.head()

Timestamp  \
0  10/14/2018 0:57:50   
1  10/14/2018 1:01:15   
2  10/14/2018 1:01:21   
3  10/14/2018 1:01:51   
4  10/14/2018 1:03:13   

   How would you rate your happiness on a scale of 0-10 with 10 being the highest?  \
0                                                  8                                 
1                                                  8                                 
2                                                  5                                 
3                                                  6                                 
4                                                  5                                 

  If you were in trouble, do you have friends or relatives you can count on to help you whenever you need them, or not?  \
0                                                Yes                                                                      
1                                                Yes                                                                      
2                                                Yes                                                                      
3                                                Yes                                                                      
4                                                Yes                                                                      

  Are you satisfied or dissatisfied with your freedom to choose what you do with your life?  \
0                                          Satisfied                                          
1                                          Satisfied                                          
2                                          Satisfied                                          
3                                          Satisfied                                          
4                                          Satisfied                                          

  Have you donated money to a charity in the past month?  \
0                                                Yes       
1                                                 No       
2                                                 No       
3                                                 No       
4                                                 No       

  Is corruption widespread throughout the government or not?  \
0                                                Yes           
1                                                Yes           
2                                                Yes           
3                                                Yes           
4                                                Yes           

  Is corruption widespread within businesses or not?  \
0                                                 No   
1                                                Yes   
2                                                Yes   
3                                                Yes   
4                                                 No   

  Did you experience any of the following during A LOT OF THE DAY yesterday? Check whichever is applicable.  \
0  Happiness?, Smiled or laughed a lot?, Enjoymen...                                                          
1  Happiness?, Smiled or laughed a lot?, Enjoymen...                                                          
2                                             Worry?                                                          
3  Happiness?, Smiled or laughed a lot?, Enjoymen...                                                          
4                                         Enjoyment?                                                          

  Do you have confidence in the national government?  \
0                                                 No   
1                                                 No   
2                                                 No   
3                                                 No   
4   

In [47]:
lifeLadder = np.mean(data.iloc[:, 1].values)
print(lifeLadder)


6.770491803278689


In [48]:
s = pd.Series(data.iloc[:, 2].values)
SocialSupport = s.value_counts(normalize =True)['Yes']
print(SocialSupport)

0.8442622950819673


In [49]:
f = pd.Series(data.iloc[:, 3].values)
Freedom = f.value_counts(normalize =True)['Satisfied']
print(Freedom)

0.75


In [50]:
g = pd.Series(data.iloc[:, 4].values)
Generosity = g.value_counts(normalize =True)['Yes']
print(Generosity)

0.29098360655737704


In [51]:
gc = pd.Series(data.iloc[:, 5].values)
GovCorruption = gc.value_counts(normalize =True)['Yes']
print(GovCorruption)

0.8934426229508197


In [52]:
bc = pd.Series(data.iloc[:, 6].values)
BusiCorruption = bc.value_counts(normalize =True)['Yes']
print(BusiCorruption)

0.8524590163934426


In [53]:
Corruption = (GovCorruption + BusiCorruption) / 2
print(Corruption)

0.8729508196721312


In [0]:
positiveAffect = 0
negativeAffect = 0
a = data.iloc[:, 7].apply(str)
positiveAffect += a.str.contains('Happiness?').sum()
positiveAffect += a.str.contains('Smiled or laughed a lot?').sum()
positiveAffect += a.str.contains('Enjoyment?').sum()
negativeAffect += a.str.contains('Worry?').sum()
negativeAffect += a.str.contains('Sadness?').sum()
negativeAffect += a.str.contains('Anger?').sum()
positiveAffect /= len(data)
negativeAffect /= len(data)

In [64]:
print(positiveAffect, negativeAffect)

1.459016393442623 1.1475409836065573


In [65]:
cg = pd.Series(data.iloc[:, 8].values)
GovtConfidence = cg.value_counts(normalize =True)['Yes']
print(GovtConfidence)

0.4426229508196721


In [67]:
t = pd.Series(data.iloc[:, 9].values)
trust = t.value_counts(normalize =True)['People can be trusted']
print(trust)

0.09836065573770492


# Evaluation metrics for SDP of Delhi
SDP per capita- 2016 = 333468 Lakhs Delhi Currency rate average in 2016 = 69.956 SDP per capita in dollars= 4766.824861341414

Log of sdp per capita 3.6782291956

Social Support = 0.84426230

Life expectancy = 73.2

Freedom to make life choices = 0.7500000

Generocity=0.29098361

Perception towards corruption(government)= 0.89344262

positive affect= 1.45901639

negative effect= 1.14754098

Confidence in national government=0.44262295

## Variables used to predict Happiness Score of Delhi

Log GDP per capita Social support Healthy life expectancy

Freedom to make life choices Generosity Perceptions of corruption

Positive affect Negative affect Confidence in national government